Common/Securities/Option/OptionStrategyPositionGroupBuyingPowerModel.cs

```diff
@@ -246,6 +246,76 @@ public override MaintenanceMargin GetMaintenanceMargin(PositionGroupMaintenanceM
 
                 return new MaintenanceMargin(inAccountCurrency);
             }
+            else if (_optionStrategy.Name == OptionStrategyDefinitions.JellyRoll.Name
+                || _optionStrategy.Name == OptionStrategyDefinitions.ShortJellyRoll.Name)
+            {
+                // long calendar spread part has no margin requirement due to same strike
+                // only the short calendar spread's short option has margin requirement
+                var furtherExpiry = parameters.PositionGroup.Positions.Max(position => position.Symbol.ID.Date);
+                var shortCalendarSpreadShortLeg = parameters.PositionGroup.Positions.Single(position => 
+                    position.Quantity < 0 && position.Symbol.ID.Date == furtherExpiry);
+                var shortCalendarSpreadShortLegSecurity = (Option)parameters.Portfolio.Securities[shortCalendarSpreadShortLeg.Symbol];
+                var result = Math.Abs(shortCalendarSpreadShortLegSecurity.BuyingPowerModel.GetMaintenanceMargin(
+                    MaintenanceMarginParameters.ForQuantityAtCurrentPrice(shortCalendarSpreadShortLegSecurity, shortCalendarSpreadShortLeg.Quantity)));
+
+                return new MaintenanceMargin(result);
+            }
+            else if (_optionStrategy.Name == OptionStrategyDefinitions.BearCallLadder.Name)
+            {
+                // Bear Call Ladder = Bear Call Spread of 2 lower strike prices + Long Naked Call with the highest strike price (margin: 0)
+                var callSpread = parameters.PositionGroup.Positions.OrderBy(position => position.Symbol.ID.StrikePrice).Take(2).ToList();
+                var shortCallSecurity = (Option)parameters.Portfolio.Securities[callSpread[0].Symbol];
+                var multiplier = Math.Abs(callSpread[0].Quantity) * shortCallSecurity.ContractUnitOfTrade;
+
+                var strikeDifference = callSpread[1].Symbol.ID.StrikePrice - callSpread[0].Symbol.ID.StrikePrice;
+                var result = strikeDifference * multiplier;
+                var inAccountCurrency = parameters.Portfolio.CashBook.ConvertToAccountCurrency(result, shortCallSecurity.QuoteCurrency.Symbol);
+
+                return new MaintenanceMargin(result);
+            }
+            else if (_optionStrategy.Name == OptionStrategyDefinitions.BearPutLadder.Name)
+            {
+                // Bear Put Ladder = Bear Put Spread of 2 lower strike prices (margin: 0) + Short Naked Put with the highest strike price
```

Comment:
Seems these option strategies are actually a combination of others, we should reuse their margin requirement calculations and avoid having to duplicate the implementations, can extract implementation into some private method/abstraction if easier